### DISCUSSION TITLE
How can I re-apply dgroups

### DISCUSSION THREAD

**Discussion by asyncmind0** [qtile/qtile#3519] [Q&A]
I want to be able to reapply dgroups after startup is this possible?

I tried using the `_add` method of `Dgroups` class 
```
def set_groups(qtile):
    from libqtile import qtile

    for client in qtile.windows_map.values()):
        logger.debug("set_groups")
        # set_group(client)
        try:
            qtile.dgroups._add(client)
        except:
            logger.exception("error setting rules %s", client)
```

**Comment by elParaguayo** [comment]
Maybe tell us what you're trying to do first ie what problem are you trying to solve?

**Reply by asyncmind0** [reply]
I just want to reapply the rules .

After I move windows to different groups, i want them to go back to pre-set groups based on dgroups
